src/ascend.ts

```diff
@@ -191,9 +206,16 @@ export function ascend(
   if (pet) visitUrl(`afterlife.php?action=buyarmory&whichitem=${toInt(pet)}`);
 
   if (permSkills) {
-    for (const [skill, permLevel] of permSkills.entries()) {
-      if (permLevel !== Lifestyle.casual) {
-        const permText = permLevel === Lifestyle.hardcore ? "hcperm" : "scperm";
+    const currentPerms = permedSkills();
+    let karma = get("bankedKarma");
+    for (const [skill, prospectivePermLevel] of permSkills.entries()) {
+      const currentPermLevel = currentPerms.get(skill) ?? Lifestyle.casual;
+      if (prospectivePermLevel > currentPermLevel) {
+        const expectedKarma = 100 * (prospectivePermLevel - currentPermLevel);
+        if (karma < expectedKarma) continue;
```

Comment:
Do we want this? With some skills being super expensive, I think it might be best to just abort perming skills when there isn't enough karma to get them all.

Reply:
We don't want to abort from inside the gash, but we also can't accurately estimate karma until before we hop the gash. It's a topic worthy of debate, but it's not as clear-cut as you may assume.